cmd/vcfContext/vcfContext.go

```diff
@@ -0,0 +1,214 @@
+// Command Group: "Variant Calling & Annotation"
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fasta"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/vcf"
+	"golang.org/x/exp/slices"
+	"log"
+	"strings"
+)
+
+func usage() {
+	fmt.Print(
+		"vcfContext - Determine the nucleotide context of mutations. \n" +
+			"Currently only works for single-nucleotide variants (e.g. T > C becomes ATG > ACG).\n" +
+			"Usage:\n" +
+			" vcfContext [options] -r ref.fa -i in.vcf > out.txt\n\n" +
+			"options:\n")
+	flag.PrintDefaults()
+}
+
+func main() {
+	var ref *string = flag.String("r", "", "Reference fasta file (must be indexed). Required for -pad > 0.")
+	var input *string = flag.String("i", "", "Input vcf file.")
+	var output *string = flag.String("o", "stdout", "Output tsv file.")
+	var includeComplements *bool = flag.Bool("includeComplements", false, "Do not merge totals from complementary mutations (e.g. separate C > A and G > T counts). Should only be used when looking at single-stranded mutations.")
+	var pad *int = flag.Int("pad", 1, "Number of context bases on either side of variant (e.g. 0 == T, 1 == ATG, 2 == TATGA, ...")
```

Comment:
Is there some kind of maximum threshold for pad, e.g. length of reference fasta sequence?

Reply:
Yes, though the memory cost of pad scales as a factorial so you will run out of memory long before you reach the length of the fasta. In reality the soft limit of pad is probably ~10, but I have not tested it.